### ISSUE TITLE
s='year' opens with disabled years when maxDate is defined

### ISSUE THREAD

**Issue by stevecastaneda** [arfedulov/semantic-ui-calendar-react#51]
**Describe the bug**
When defining a maxDate and startMode='year', calendar should open with maxDate year as the last option, and all previous years available for immediate selection vs. having to click on the arrow to view the next series of years.

**To Reproduce**
1. Create a DateInput.
2. Define startMode as 'year'.
3. Define maxDate as 'moment()'

**Expected behavior**
Years available will only show a few options, 16/17/18 as of today, and grey out all the future options as expected due to maxDate, but instead should open with 18' as the last year with all previous years as options.

**Screenshots**
https://www.dropbox.com/s/fc6ooev4o268sxn/Screenshot%202018-12-03%2014.49.04.png?dl=0

**Dependencies versions**
Provide version numbers of following packages:

* semantic-ui-react - 0.83.0
* semantic-ui-css (or any alternative) N/A
* semantic-ui-calendar-react - 0.12.1

**Comment by arfedulov** [comment]
I've done the steps for reproducing the bug you've described. I have this code:
```jsx
<DateInput
          startMode="year"
          popupPosition="bottom right"
          placeholder="Date startMode year"
          className="example-calendar-input"
          name="dateStartYear"
          value={this.state.dateStartYear}
          iconPosition="left"
          maxDate={moment()}
          autoComplete="off"
          preserveViewMode={false}
          onChange={this.handleChange} />
```

And I see this:
![scr](https://user-images.githubusercontent.com/19156819/49482596-e2b66600-f840-11e8-9438-da78d7ce189a.png)

I probably missed something because for now it seems like the calendar works as expected.

**Comment by stevecastaneda** [comment]
Thanks for checking. The goal was to instead have 2018 show up as the last selectable item as there are no results eligible beyond it, instead of showing ineligible years.

**Comment by stevecastaneda** [comment]
Exactly!  It's more of a UX issue for sure, but I think it's helpful when maxDate.  Appreciate you looking into it.

**Comment by dopeboy** [comment]
Are there any workarounds for this?
